tests/console/zypper_lifecycle.pm

```diff
@@ -34,16 +34,28 @@ use warnings;
 use testapi;
 use utils;
 use version_utils qw(is_sle is_jeos);
+use Utils::Architectures;
 
 our $date_re = qr/[0-9]{4}-[0-9]{2}-[0-9]{2}/;
 
 sub run {
     diag('fate#320597: Introduce \'zypper lifecycle\' to provide information about life cycle of individual products and packages');
+
+    # Workaround for bsc#1166549, zypper lifecycle error with 'Error building the cache'.
+    # Currently the only workaround option is to run zypper-lifecycle as root.
+
+    # Now we just hit bsc#1166549 on s390x and ppc64le, so we just need root console
+    # on these two platforms.
+    my $needs_root_console = (check_var('ARCH', 's390x') || check_var('ARCH', 'ppc64le'));
+    if ((is_s390x || is_ppc64le) && is_sle('>=15-SP3') && get_var('UPGRADE')) {
```

Comment:
It is better to use is_upgrade instead of get_var('UPGRADE').

Reply:
Changed, thanks